### PR TITLE
[Mousetrap] Add mousetrap

### PR DESCRIPTION
**Pull request by Clemapfel**
Supercedes #7347. 

I neglected to expand for C++ string ABIs because my project requires gcc 11.0 or newer anyway, so there is no way to build with the old standard. Please correct me if this is improper.

### CODE REVIEW

**Comment by Clemapfel** [comment]
I had to add the following line to the `MESON_TARGET_TOOLCHAIN` in the `[binaries]` section:

```ini
make='/usr/bin/cmake'
```

Otherwise it would not detect the cmake executable for me. Should I open an issue for this?

File: M/mousetrap/build_tarballs.jl
Comment (imciner2) [inline M/mousetrap/build_tarballs.jl:39]: Are you just trying to not use any 32-bit platforms? (I note that this filter will leave only the 64-bit platforms).

**Review by imciner2 — CHANGES_REQUESTED** [review]

File: M/mousetrap/build_tarballs.jl
Comment (Clemapfel) [reply M/mousetrap/build_tarballs.jl:39]: yes, exactly. I couldn't find a is64bit function so I did it with pattern matching

Comment (imciner2) [reply M/mousetrap/build_tarballs.jl:36]: You can use `nbits(platform)` to find out the platform, so to only get the 64-bit platforms you can use the comparison `nbits(platform) == 64`.

Comment (Clemapfel) [reply M/mousetrap/build_tarballs.jl]: that is much nicer, fixed by https://github.com/JuliaPackaging/Yggdrasil/pull/7461/commits/e6ef38bb1477de1a55f5e14193a74ffe091d61f1

**Review by staticfloat — APPROVED** [review]

File: M/mousetrap/build_tarballs.jl
Comment (giordano) [inline M/mousetrap/build_tarballs.jl:59]: Please don't use gcc 12, you should instead use the oldest compiler possible: https://github.com/JuliaPackaging/Yggdrasil/blob/7093500a13b031888e5263aa3fcf7c63490bbe2b/CONTRIBUTING.md#compatibility-tips

**Review by giordano — COMMENTED** [review]
> I neglected to expand for C++ string ABIs because my project requires gcc 11.0 or newer anyway, so there is no way to build with the old standard. Please correct me if this is improper.

Quoting from the [docs](https://docs.binarybuilder.org/stable/building/#Expanding-C-string-ABIs-or-libgfortran-versions)

> This ABI does _not_ have to do with the C++ standard used by the source code, in fact you can build a C++03 library with the C++11 `std::string` ABI and a C++11 library with the C++03 `std::string` ABI. This is achieved by appropriately setting the `_GLIBCXX_USE_CXX11_ABI` macro.

Emphasis not mine.

File: M/mousetrap/build_tarballs.jl
Comment (Clemapfel) [reply M/mousetrap/build_tarballs.jl]: I added two commits to expand for string ABIs and to use GCC 9, which is the oldest version supported. Thank you

**Comment by stemann** [comment]
Is support for Julia 1.7 and 1.8 needed? You are spending a lot of CI hours for building this...

**Comment by Clemapfel** [comment]
I agree, but [I was told by fingolfin](https://github.com/JuliaPackaging/Yggdrasil/pull/7347#discussion_r1328071397) to expand for different Julia versions. This and the string ABI expansion makes for a ridiculous amount of versions.

If I only use Julia 1.9, will that mean 1.8 and 1.7 users cannot use the jll package? I'd like to not introduce a version dependency when there isn't any reason for it other than for build time.

**Comment by stemann** [comment]
> I agree, but [I was told by fingolfin](https://github.com/JuliaPackaging/Yggdrasil/pull/7347#discussion_r1328071397) to expand for different Julia versions. This and the string ABI expansion, both of which I didn't have initially but were requested during review, makes for a ridiculous amount of versions.
> 
> If I only use Julia 1.9, will that mean 1.8 and 1.7 users cannot use the jll package? I'd like to not introduce a version dependency when there isn't any reason for it other than for build time.

Indeed.

Yes - that would drop support for 1.8 and 1.7 users.

**Comment by Clemapfel** [comment]
I would be fine to change:
```julia
platforms = Platform[]
include("../../L/libjulia/common.jl")
for version in [v"1.7.0", v"1.8.2", v"1.9.0", v"1.10", v"1.11"]
    for platform in libjulia_platforms(version)
        if nbits(platform) != 32
            push!(platforms, platform)
        end
    end
end
platforms = expand_cxxstring_abis(platforms)
```
To 
```
platforms = expand_cxxstring_abis(filter!(p -> nbits != 64, supported_platforms())
```
Since I'm not positive any of the Julia api changes between versions affect my project, but I would delegate that decision to you guys since I don't have any experience with deploying jlls.

**Comment by Clemapfel** [comment]
I copied the versions expansion mostly form https://github.com/JuliaPackaging/Yggdrasil/blob/cefcd399d9ef68bced747dd2d3c9ec59f32d169f/L/libcxxwrap_julia/build_tarballs.jl which also has that many versions and is already merged.

**Comment by ufechner7** [comment]
@Clemapfel Are you still waiting for further input, or do you think it can be merged?

**Comment by ufechner7** [comment]
@imciner2 Can you please approve?

**Comment by Clemapfel** [comment]
@imciner2 is there something else I can do? I'm planning on opening a PR with the Julia registry for a package that depends on this jll, but I cannot do so until this is merged and online.

**Review by imciner2 — APPROVED** [review]

**Comment by imciner2** [comment]
This is good now. It should appear in the registry in a few hours.

**Comment by Clemapfel** [comment]
thank you!

**Pull request merged**
